properties.md

```diff
@@ -0,0 +1,140 @@
+
+```json5
+{
+    "type":  "olm.package",
+    "value": `{"packageName":"etcd","version":"0.9.2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdCluster","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdRestore","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdBackup","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.label",
+    "value": `{"label":"testlabel"}`,
+}
+```
+
+Properties intentionally closely align with the format for [`dependencies.yaml`](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-dependencies).
+
+Today, all properties are derived from other fields in a bundle. There is no way to write a new property `type` for a consumer (say, a newer version of the resolver, or an external tool that processes sets of bundles).
+
+### Other consumers
+
+Catalog and Bundles are consumed in many other contexts than the resolver. The package-server component presents catalog content for the purpose of selecting packages and channels, for example. 
+
+### Goals
+
+- Define the format for explicitly adding properties to a bundle.
+
+## Proposal
+
+A new file is accepted in the `metadata` folder for a bundle: `properties.(yaml|json)`
+
+Here is an example of `properties.yaml`:
+
+```yaml=
+properties:
+- type: olm.package
+  value:
+    packageName: prometheus
+    version: "0.27.0"
+- type: olm.gvk
+  value:
+    group: etcd.database.coreos.com
+    kind: EtcdCluster
+    version: v1beta2
+- type: olm.label
+  value: "LTS"
+```
+
+If properties are defined, they take precendence over the values derived from the bundle contents. In the above example, the `version` of the bundle will be considered `0.27.0` regardless of what the `ClusterServiceVersion`'s `spec.version` field indicates.
+
+### Validate properties
+
+Property validation is the domain of the consumer, not the producer. 
+
+However, `opm` will ship with validation for all properties that the OLM resolver understands, and they will be validated as part of the `opm alpha bundle validate` command and the `operator-sdk` bundle validation. 
+
+A future enhancement will discuss a formalized way to discover / publish validation rules for a given property `type`.
+
+### Generate default properties
+
+`opm` and `operator-sdk` will be capable of scaffolding the set of inferred properties from a given bundle. `opm` today infers a set of properties based on bundle content and stores them in the registry.
+
+### Initial Resolver-Consumer Properties
+
+This is a list of properties that the resolver consumer will read and understand.
+
+- `olm.package`: The package name and bundle version. Used as a target of version range dependencies.
```

Comment:
Does this attempt to answer the question of bundle identity? Or is that not a goal of this enhancement?

Reply:
That's not a goal of the enhancement - that would be an example of consumer-defined semantics. I would expect that to be defined in the declarative config enhancement (if not just documentation of what the resolver chooses to do in the face of properties)

Reply:
That's fair, I'm just wondering if we want to delineate those as special "first-class properties" that, when not included, imply you do not have a valid bundle. Or should we have a notion that that would never be the case, and property requirement is defined solely by bundle type?

Reply:
The way I see it, if something is a first-class property then it's not a property at all according to this proposal.